docs/CSL-1925.md

```diff
@@ -0,0 +1,256 @@
+
+# Problem Definition
+
+An address in Cardano is always derived from a root key using BIP32 style derivation
+except it uses a different elliptic curve than Bitcoin (Ed25519). 
+
+The problem initially raised concerns exchanges which have only one account per root
+key and thus 2^31 possible addresses. In this scenario according to the birthday
+paradox there is a non negligible chance of a collision occuring.
+
+The first child (or branch) of the root key is derived according to the _account index_
+which by default is 2^31 or in binary `10000000000000000000000000000000` where the
+first bit represents the hardened state. As a BIP32 path this can be represented as 
+`m/aH`. The second child is derived according to an _address index_.
+
+Both the account and address indices fall within the range 2^31 in terms of how many
+addresses can be derived.
+
+In the current scheme the address size is 76 bytes, where each index takes up 5 bytes.
+
+# Generation Strategies
+
+There are currently two ways of generating a unique address, as depicted by the following
+ADT.
+
+```haskell
+data GenerationMode 
+  = RandomMode
+  -- ^ Generates an address where the second index is random and hardened.
+  | DeterministicMode Word32 
+  -- ^ Generates an address where the second index is parameterised.
+```
+
+The difference between `RandomMode` and `DeterministicMode` is that in `DeterministicMode` the
+application / dependent code or user must supply the seed used to generate the index instead 
+of it being generated randomly. `DeterministicMode` is there in order to be able to recover 
+the address at a later date.
```

Comment:
Minor bikeshedding, why not saying "[...]`DeterministicMode` is used during the wallet recovery process at a later date[..]"? This way, we give that `DeterministicMode` a purpose without necessarily implying that this is a necessity for recover (as far as we know, this might or might not be true). Mental?